### PR TITLE
fetch and save identities scheduled

### PR DESCRIPTION
**Pull request by s-dobschal**

### CODE REVIEW

File: src/repositories/identityRepository.ts [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
+                    )
+                    VALUES ? on duplicate key update 
+                        active = values(active),
+                        display = values(display),
+                        address = values(address),
+                        riot = values(riot),
+                        twitter = values(twitter),
+                        web = values(web),
+                        email = values(email);`;
+        const data = [identities.map((identity)=> [accounts?.find(account => account.address === identity.basicInfo.address)?.id, true, identity.basicInfo.display, identity.basicInfo.legal, identity.basicInfo.address, identity.basicInfo.riot, identity.basicInfo.twitter, identity.basicInfo.web, identity.basicInfo.email])];
```
Comment (dobschal) [inline src/repositories/identityRepository.ts:33]: @a-oertel please format correctly. That map/find looks weird.
Screenshot 2022-05-23 at 08 40 34

**Review by dobschal — CHANGES_REQUESTED** [review]

File: src/repositories/accountRepository.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+    async findAllByChainId(chain_id: number): Promise<AccountEntity[]|undefined> {
+        const query = `SELECT *
+                       FROM ${this.tableName} 
+                       WHERE ${this.tableName}.chain_id=${chain_id}`;
+        return (await runSelectQuery<AccountEntity>(query));
+    }
+
+    async insertOrUpdateAccountsOfIdentities(identities: Identity[], chainId: number): Promise<QueryResult> {
+        const data = [identities.map((identity)=> [identity.basicInfo.address, chainId])];
```
Comment (dobschal) [inline src/repositories/accountRepository.ts:19]: @a-oertel data is then an array with one item. That one item is an array containing arrays.. Is that correct?

```javascript
const data = [
    [
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ["{address}", "{chainId}"],
        ...
    ]
]
```

**Review by dobschal — CHANGES_REQUESTED** [review]

File: src/repositories/identityRepository.ts [hunk 3]
```diff
@@ -1,10 +1,39 @@
...
+        return (await runInsertQuery(query, data));
```
Comment (dobschal) [inline src/repositories/identityRepository.ts:34]: @a-oertel  there is no update query yet

Comment (a-oertel) [reply src/repositories/identityRepository.ts:34]: it updates with `on duplicate key update`

File: src/repositories/accountRepository.ts (hunk 2 above)
Comment (a-oertel) [reply src/repositories/accountRepository.ts]: It is working as expected like that

File: src/dbMigrations/0002_add_account_table.sql [hunk 4]
```diff
@@ -3,5 +3,6 @@ CREATE TABLE account (
     chain_id INT(11) UNSIGNED NOT NULL,
     FOREIGN KEY(chain_id) REFERENCES chain(id),
     address VARCHAR(255),
+    CONSTRAINT address_chain UNIQUE (address,chain_id),
```
Comment (dobschal) [inline src/dbMigrations/0002_add_account_table.sql:6]: @a-oertel add separate migration script

**Review by dobschal — CHANGES_REQUESTED** [review]

**Review by dobschal — APPROVED** [review]

**Pull request merged**